source/Octopus.TestPortForwarder/SocketPump.cs

```diff
@@ -69,15 +91,34 @@ async Task PausePump(CancellationToken cancellationToken)
                 cancellationToken.ThrowIfCancellationRequested();
             }
         }
-
-        static async Task WriteToSocket(Socket writeTo, byte[] inputBuffer, int totalBytesToSend, CancellationToken cancellationToken)
+        
+        async Task WriteToSocketDelayingSendingTheLastNBytes(Socket writeTo, byte[] buffer, int bufferLength, int delaySendingLastNBytes, CancellationToken cancellationToken)
+        {
+            int howMuchToSend = bufferLength - delaySendingLastNBytes;
+            if(howMuchToSend < 0) howMuchToSend = bufferLength;
+            int sent = await WriteToSocket(writeTo, buffer, 0, howMuchToSend, cancellationToken);
+            if (howMuchToSend < bufferLength)
+            {
+                await Task.Delay(10);
+                int restToSend = bufferLength - howMuchToSend;
+                sent += await WriteToSocket(writeTo, buffer, howMuchToSend, bufferLength - howMuchToSend, cancellationToken);
+            }
+            if (sent != bufferLength)
+            {
+                throw new Exception($"Was supported to send {bufferLength} but sent {sent}");
+            }
+        }
+        
+        static async Task<int> WriteToSocket(Socket writeTo, byte[] buffer, int initialOffset, int totalBytesToSend, CancellationToken cancellationToken)
         {
-            var offset = 0;
+            ArraySegment<byte> toSend = new ArraySegment<byte>(buffer, initialOffset, totalBytesToSend);
```

Comment:
Just wondering, instead of introducing the `Slice` below, could we not simply use the code as it was, and set `offset` to `initialOffset`?

I'm likely missing a subtlety here...

Reply:
I originally didn't use slice but the code was cleaner with slice, then later I learned slice was not a net48 thing so I added the extension below to keep this method simpler.